### PR TITLE
improve state management for vector layers

### PR DESCRIPTION
**Pull request by Thuhaa**

### CODE REVIEW

**Comment by Thuhaa** [comment]
Fixes #1122 
Fixes #1130

**Comment by Thuhaa** [comment]
**State management for the different vector layer components is improved**
1. `PropertySelect.svelte` component in the `VectorLegendAdvanced` image

  2. `numberOfClasses, and classificationMethod` and therefore the `colorMapRows` state is maintained 
image

File: sites/geohub/src/components/VectorLayer.svelte [hunk 1]
```diff
@@ -4,20 +4,25 @@
...
+  let applyToOption: VectorApplyToTypes = VectorApplyToTypes.COLOR
   let legendType: string
   let defaultColor: string
   let defaultLineColor: string
-
-  let activeTab = ''
+  let classification: ClassificationMethodTypes = classificationMethod
+  let cMapName: string = colorMapName
+  let numberOfClasses: number = COLOR_CLASS_COUNT
+  let colorMapRows: Array<IntervalLegendColorMapRow> = []
```
Comment (JinIgarashi) [inline sites/geohub/src/components/VectorLayer.svelte:24]: we don't need to keep `numberOfClasses` and `colorMapRows` in this level.

File: sites/geohub/src/components/controls/VectorLegendAdvanced.svelte [hunk 2]
```diff
@@ -40,38 +40,38 @@
...
+  export let applyToOption: VectorApplyToTypes
   export let layer: Layer
   export let layerMax: number
   export let layerMin: number
   export let colorMapName: string
   export let defaultColor: string
+  export let propertySelectValue
+  export let numberOfClasses: number
+  export let colorMapRows: IntervalLegendColorMapRow[]
+  export let defaultOutlineColor: string
```
Comment (JinIgarashi) [inline sites/geohub/src/components/controls/VectorLegendAdvanced.svelte:52]: these four properties may not need to be exported. and all  information can be extracted from layer style object

**Review by JinIgarashi — CHANGES_REQUESTED** [review]
@Thuhaa Don't add unnecessary props which can be extracted from maplibre layer style object.

File: sites/geohub/src/components/controls/VectorLegendPanel.svelte [hunk 3]
```diff
@@ -187,11 +185,11 @@
           bind:layer
           bind:layerMin
           bind:layerMax
+          bind:propertySelectValue
```
Comment (JinIgarashi) [inline sites/geohub/src/components/controls/VectorLegendPanel.svelte:188]: delete this. this property is not exported

**Review by JinIgarashi — CHANGES_REQUESTED** [review]
The colormap picker looks different from before??

image

**Review by JinIgarashi — APPROVED** [review]

**Pull request merged**